tests/test_get_notebooks.py

```diff
@@ -0,0 +1,27 @@
+"""Check function which lists notebooks in directory."""
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from nbqa.__main__ import _get_notebooks
+
+if TYPE_CHECKING:
+    from _pytest.monkeypatch import MonkeyPatch
+
+
+def test_get_notebooks(monkeypatch: "MonkeyPatch"):
+    """
+    Check that unwanted directories are excluded.
+
+    Parameters
+    ----------
+    monkeypatch
+        Pytest fixture, we use it to override ``PYTHONPATH``.
+
+    """
+    result = list(_get_notebooks("tests"))
+    assert Path("tests") / "data/clean_notebook.ipynb" in result
+
+    monkeypatch.setattr("nbqa.__main__.EXCLUDES", r"/(data)/")
```

Comment:
This does not really test your EXCLUDES on code that will be shipped. Consider the scenario of having a bug in the `EXCLUDES` regex pattern. I would recommend to use `def test_get_notebooks(tmpdir: "LocalPath")`. Create `.git` dir inside this temp directory. Copy the notebook from `tests/data/clean_notebook.ipynb` to `tmpdir/.git`. You could do the same for many other directories like venv etc.

Reply:
yes, that would be better, thanks!